factory/runType/sovereignRunTypeComponents.go

```diff
@@ -18,21 +19,24 @@ import (
 	"github.com/multiversx/mx-chain-go/process/sync"
 	"github.com/multiversx/mx-chain-go/process/sync/storageBootstrap"
 	"github.com/multiversx/mx-chain-go/process/track"
+	"github.com/multiversx/mx-chain-go/state/factory"
 	storageFactory "github.com/multiversx/mx-chain-go/storage/factory"
 )
 
 type sovereignRunTypeComponentsFactory struct {
 	*runTypeComponentsFactory
+	cfg config.SovereignConfig
 }
 
 // NewSovereignRunTypeComponentsFactory will return a new instance of runTypeComponentsFactory
-func NewSovereignRunTypeComponentsFactory(fact *runTypeComponentsFactory) (*sovereignRunTypeComponentsFactory, error) {
+func NewSovereignRunTypeComponentsFactory(fact *runTypeComponentsFactory, cfg config.SovereignConfig) (*sovereignRunTypeComponentsFactory, error) {
 	if check.IfNil(fact) {
 		return nil, errors.ErrNilRunTypeComponentsFactory
 	}
 
```

Comment:
or maybe test `NativeESDT` here to be covered in unit tests

Reply:
Will add a task for this to be done whenever we also have token prefix.